models/orderDao.js

```diff
@@ -0,0 +1,82 @@
+const appDataSource = require('./appDataSource');
+const { v4: uuid } = require('uuid');
+
+const createOrder = async (userId, totalPrice) => {
+  const queryRunner = appDataSource.createQueryRunner();
+  await queryRunner.connect();
+  await queryRunner.startTransaction();
+  try {
+    const orderNum = uuid();
+    await queryRunner.query(
+      `INSERT INTO orders(
+        user_id,
+        total_price,
+        number,
+        address_id
+        )
+        SELECT ?, ?, ?, addresses.id
+        FROM addresses
+        WHERE user_id = ?
+        ORDER BY id DESC LIMIT 1
+        `,
+      [userId, totalPrice, orderNum, userId]
+    );
+    const orderItems = async (userId) => {
+      await queryRunner.query(
+        `INSERT INTO order_items(
+        order_id ,
+        product_id ,
+        quantity ,
+        status_id
+        )
+        SELECT orders.id, carts.product_id, carts.quantity , orders.status_id
+        FROM orders
+        JOIN carts ON carts.user_id = orders.user_id
+        WHERE orders.user_id = ? AND orders.status_id = 1
+      `,
+        [userId]
+      );
+    };
+    await orderItems(userId);
```

Comment:
굳이 함수 내의 함수로 만들어 실행하는 것이 아닌, 하나의 query로 작성되는 것이 깔끔할 것 같습니다.
```suggestion
      await queryRunner.query(
        `INSERT INTO order_items(
        order_id ,
        product_id ,
        quantity ,
        status_id
        )
        SELECT orders.id, carts.product_id, carts.quantity , orders.status_id
        FROM orders
        JOIN carts ON carts.user_id = orders.user_id
        WHERE orders.user_id = ? AND orders.status_id = 1
      `,
        [userId]
      );
```